presto-cli/src/main/java/com/facebook/presto/cli/AlignedTablePrinter.java

```diff
@@ -58,6 +58,13 @@ public void finish()
     {
         printRows(ImmutableList.<List<?>>of(), true);
         writer.append(format("(%s row%s)%n", rowCount, (rowCount != 1) ? "s" : ""));
+        flush();
+    }
+
+    @Override
+    public void flush()
```

Comment:
I think that the final `printRows()` call should be moved here, and `flush()` should only be called once on failure (and internally by `finish()`).